sequencesender/sequencesender.go

```diff
@@ -1168,7 +1168,8 @@ func (s *SequenceSender) addInfoSequenceBatchEnd(batch *datastream.BatchEnd) {
 // addNewBatchL2Block adds a new L2 block to the work in progress batch
 func (s *SequenceSender) addNewBatchL2Block(l2Block *datastream.L2Block) {
 	s.mutexSequence.Lock()
```

Comment:
I'm aware that it is unrelated to this PR, but I'm afraid that we are not releasing the lock in case we are returning from the function (https://github.com/0xPolygon/cdk/pull/86/files#diff-9420435b2f8e7ca901e1efd3c5f0dd6c7e267d8a90ae837c135fbd8f239ea5bfR1199-R1202) or even in case we are logging on fatal level (https://github.com/0xPolygon/cdk/pull/86/files#diff-9420435b2f8e7ca901e1efd3c5f0dd6c7e267d8a90ae837c135fbd8f239ea5bfR1181-R1184). So if you don't mind, probably it would be a better idea to unlock on `defer` statement instead of at the end of the function.

```suggestion
	s.mutexSequence.Lock()
	defer s.mutex.Sequence.Unlock()
```

Reply:
good catch, I have also changed any unlock with a defer one